src/panels/lovelace/editor/hui-dialog-edit-card.ts

```diff
@@ -3,9 +3,33 @@ import { TemplateResult } from "lit-html";
 
 import { HomeAssistant } from "../../../types";
 import { LovelaceCardConfig } from "../types";
+import { EditDialogParams } from "./types";
+import { fireEvent } from "../../../common/dom/fire_event";
 import "./hui-edit-card";
 import "./hui-migrate-config";
 
+const dialogShowEvent = "show-edit-card";
+const dialogTag = "hui-dialog-edit-config";
+
+export const registerEditCardDialog = (element: HTMLElement) =>
+  fireEvent(element, "register-dialog", {
+    dialogShowEvent,
+    dialogTag,
+    dialogImport: () => import("./hui-dialog-edit-card"),
+  });
+
+export const showEditCardDialog = (
+  element: HTMLElement,
+  hass: HomeAssistant,
+  cardConfig: LovelaceCardConfig,
+  reloadLovelace: () => void
+) =>
+  fireEvent(element, dialogShowEvent, {
+    hass,
+    cardConfig,
+    reloadLovelace,
+  });
+
 export class HuiDialogEditCard extends LitElement {
   protected _hass?: HomeAssistant;
```

Comment:
No need for this: https://github.com/home-assistant/home-assistant-polymer/blob/101a364a83d2a20fa472e88a86102b037ed0f3d2/src/layouts/app/dialog-manager-mixin.js#L16-L18